dnf5daemon-server/session.cpp

```diff
@@ -368,18 +376,10 @@ void Session::store_transaction_offline() {
     }
 
     // Serialize the transaction
-    const auto & offline_datadir = installroot / libdnf5::offline::DEFAULT_DATADIR.relative_path();
-    std::filesystem::create_directories(offline_datadir);
-
     constexpr const char * packages_in_trans_dir{"./packages"};
     constexpr const char * comps_in_trans_dir{"./comps"};
     const auto & comps_location = offline_datadir / comps_in_trans_dir;
 
-    const std::filesystem::path state_path{offline_datadir / libdnf5::offline::TRANSACTION_STATE_FILENAME};
-    libdnf5::offline::OfflineTransactionState state{state_path};
-
-    auto & state_data = state.get_data();
-
     state_data.set_status(libdnf5::offline::STATUS_DOWNLOAD_INCOMPLETE);
     state.write();
```

Comment:
Why is the state written twice? Did I miss something being changed in it?

Reply:
Good catch! The intention was to move setting `STATUS_DOWNLOAD_INCOMPLETE` before the download is initiated. I reality I only copied the lines. Will fix it.